src/iceberg/test/mock_catalog.h

```diff
@@ -23,6 +23,7 @@
 #include <gtest/gtest.h>
 
 #include "iceberg/catalog.h"
+#include "iceberg/table.h"
```

Comment:
Fix a bug. Otherwise clang-tidy checks would fail when someone else modified this file. 
The reasion is `MOCK_METHOD((Result<std::unique_ptr<Table>>), ...` will instantiate `std::expected<std::unique_ptr<Table>>` but `Table` here is just a forward declaration so `std::unique_ptr` will fail to generate destructor.

Reply:
If we change `std::unique_ptr<Table>` to `std::shared_ptr<Table>` like https://github.com/apache/iceberg-cpp/pull/418, we don't need to get the complete definition.